src/net_processing.cpp

```diff
@@ -3881,8 +3883,6 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
     {
         LOCK(cs_main);
 
-        if (MaybeDiscourageAndDisconnect(*pto)) return true;
```

Comment:
edd1a8a5c76a15174985fe0a1a876eba9e7ee33c

Do you have an idea why this wasn't in the top?

Reply:
Because it used to be `SendRejectsAndCheckIfBanned()`, and before that it was just sending rejects. Those required cs_main, which is taken after the ping processing at the top.